pkg/dns/service.go

```diff
@@ -60,10 +63,10 @@ func (s *Service) resolveIPS(ctx context.Context, addresses []gatewayv1beta1.Gat
 }
 
 func (s *Service) GetManagedHosts(ctx context.Context, traffic traffic.Interface) ([]v1alpha1.ManagedHost, error) {
-	managed := []v1alpha1.ManagedHost{}
+	var managed []v1alpha1.ManagedHost
 	for _, host := range traffic.GetHosts() {
 		managedZone, subDomain, err := s.GetManagedZoneForHost(ctx, host, traffic)
-		if err != nil {
+		if err != nil && !errors.Is(err, ErrNoManagedZoneForHost) {
```

Comment:
We need to ignore these "no managed zone for host" errors or it will only ever check the first one